e2etests/e2etests.js

```diff
@@ -41,12 +41,16 @@ device_provision(hubConnectionString, function (err, provisionedDevices) {
     console.log('Unable to create the devices needed.');
   } else {
     provisionedDevices.forEach(function(deviceToTest) {
-      acknowledgementProtocols.forEach(function (protocolToTest) {
-        device_acknowledge_tests(hubConnectionString, protocolToTest, deviceToTest);
+      if (deviceToTest.authenticationDescription !== 'shared private key') {
```

Comment:
The test matrix is becoming more complicated and by adding this kind of conditions we make it harder to maintain. The pattern that separates device provisioning from the tests themselves is starting to show cracks.

also, if we were to stick with this, string description sounds like a bad choice to solve a type problem - enum values would work better.